### ISSUE TITLE
Make Docker supervisord not run programs as root so it plays nice with Kubernetes

### ISSUE THREAD

**Issue by TimLethbridge** [umple/umple#1613]
Users trying to make the UmpleOnline docker image run with Kubernetes have been experiencing problems because umpleonline/docker_config/supervisord.conf specifies user=root

It needs to run as a user other than root.

The solution might be to simply delete that line, or change to user=www. It might also be necessary to change to group=www too

Additional changes may be needed to umpleonline/docker_config/nginx.conf and/or php-fpm.conf
A change to Dockerfile-base might be needed to add a user (possibly)

Some documentation to help with this can be found at the following sites:

Supervisord configuration: http://supervisord.org/configuration.html

Suggestion to use www or possibly webuser or possibly www-date: https://serverfault.com/questions/433265/how-do-i-change-the-nginx-user

Current supervisord log says: 
```

2020-07-28 16:38:23,642 CRIT Can't drop privilege as nonroot user

2020-07-28 16:38:23,644 INFO supervisord started with pid 8

2020-07-28 16:38:24,648 INFO spawned: 'nginx' with pid 11

2020-07-28 16:38:24,652 INFO spawned: 'php-fpm' with pid 12

2020-07-28 16:38:24,660 DEBG 'nginx' stderr output:

2020/07/28 16:38:24 [warn] 11#11: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2

nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2

2020-07-28 16:38:24,662 DEBG fd 9 closed, stopped monitoring <POutputDispatcher at 139664490379528 for <Subprocess at 139664490378880 with name php-fpm in state STARTING> (stdout)>

2020-07-28 16:38:24,662 DEBG 'nginx' stderr output:

2020/07/28 16:38:24 [emerg] 11#11: bind() to 0.0.0.0:80 failed (13: Permission denied)

nginx: [emerg] bind() to 0.0.0.0:80 failed (13: Permission denied)

2020-07-28 16:38:24,662 DEBG fd 8 closed, stopped monitoring <POutputDispatcher at 139664490378952 for <Subprocess at 139664490378736 with name nginx in state STARTING> (stderr)>

2020-07-28 16:38:24,662 DEBG fd 6 closed, stopped monitoring <POutputDispatcher at 139664504118880 for <Subprocess at 139664490378736 with name nginx in state STARTING> (stdout)>

2020-07-28 16:38:24,663 INFO exited: nginx (exit status 1; not expected)

2020-07-28 16:38:24,663 DEBG received SIGCHLD indicating a child quit

2020-07-28 16:38:24,687 DEBG 'php-fpm' stderr output:

[28-Jul-2020 16:38:24] NOTICE: [pool umple] 'user' directive is ignored when FPM is not running as root

2020-07-28 16:38:24,688 DEBG 'php-fpm' stderr output:

[28-Jul-2020 16:38:24] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root

[28-Jul-2020 16:38:24] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root

2020-07-28 16:38:24,688 DEBG 'php-fpm' stderr output:

[28-Jul-2020 16:38:24] ERROR: unable to bind listening socket for address '/var/run/php-fpm.sock': Permission denied (13)

[28-Jul-2020 16:38:24] ERROR: FPM initialization failed

2020-07-28 16:38:24,688 DEBG fd 14 closed, stopped monitoring <POutputDispatcher at 139664490379240 for <Subprocess at 139664490378880 with name php-fpm in state STARTING> (stderr)>

2020-07-28 16:38:24,688 INFO exited: php-fpm (exit status 78; not expected)

2020-07-28 16:38:24,689 DEBG received SIGCHLD indicating a child quit
...

**Comment by TimLethbridge** [comment]
This may require some experimentation. A new Docker image will be created for the branch that is created for this change, so test that image.

**Comment by TimLethbridge** [comment]
Closed in pull request #1616
